### PR TITLE
Support for virtual execution module pkg on AIX

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Added support for virtual execution module pkg on AIX,
provides support for pkg.install, pkg.remove, pkg.list_pkgs, pkg.version

Also adds support for pkg.is_installed

Adds ability to install/remove IBM AIX filesets and/or rpm packages for the AIX platform

### What issues does this PR fix or reference?
https://github.com/saltstack/zh/issues/1408

### Previous Behavior
There was no support for execution module pkg on AIX

### New Behavior
pkg.install, pkg.remove, pkg.list_pkgs, pkg.version, pkg.is_installed are now functional on AIX

### Tests written?
No - Tests coming lates

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by terminalmage — CHANGES_REQUESTED** [review]
This is missing functions like `latest_version`. Please refer to our package provider writing documentation here: https://docs.saltstack.com/en/latest/topics/development/package_providers.html#package-functions

**Comment by dmurphy18** [comment]
@terminalmage I know, talked with Kent Besaw about the additional functionality, since he talks with the customer weekly (that requested this support) and agreed to limit to the functionality requested rather than adding extra's not originally requested. So pulled that out: but can add it and others back in to provide barest minimum support to meet Fluorine commitments.  Can add latest_available and upgrade_available (not too applicable in AIX since no repositories as in the sense of Linux or even Solaris, and the latest is that already committed to the database AIX keeps - ODM). Can add the functions and have them return what is already there.

**Comment by terminalmage** [comment]
I don't find that to be an acceptable solution, since pkg states expect at least some of the missing functions to exist and will throw a KeyError if you run them and they're not.

**Comment by dmurphy18** [comment]
@terminalmage I was going to check the states pkg to see what was affected when writing the unit tests. I shall have another talk with Kent and raise the issues again with him, to push for the other functionality.

**Comment by dmurphy18** [comment]
@terminalmage talked with Kent and adding latest_version and upgrade_available. Others shall be added in another PR, to ensure this meets deadlines

**Comment by dmurphy18** [comment]
@terminalmage Added support for pkg.latest_version, pkg.upgrade_available similar to Solaris 10 which also does not support repositories.

File: salt/modules/aixpkg.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.latest_version <package name>
+        salt '*' pkg.latest_version <package1> <package2> <package3> ...
+
+    NOTE: As package repositories are not presently supported for AIX
```
Comment (terminalmage) [inline salt/modules/aixpkg.py]: "package repositories" is redundant as it results in the word "package" appearing twice in the same clause. Also, the 2nd appearance should be "packages", not "package". Thus, this should be:

```
    NOTE: As repositories are not presently supported for AIX installp/RPM
    packages, this function will always return an empty string for a given
    package.
```

**Review by terminalmage — CHANGES_REQUESTED** [review]

File: salt/modules/aixpkg.py (hunk 1 above)
Comment (dmurphy18) [reply salt/modules/aixpkg.py]: @terminalmage I hate getting nailed by cut and paste, :(, will fix.

**Comment by dmurphy18** [comment]
@rallytime Those pylint errors are fixed.

**Comment by dmurphy18** [comment]
@rallytime Can Mike be pulled as a reviewer since he is PTO and a continent and an ocean away ?

File: salt/modules/aixpkg.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+
+def __virtual__():
+    '''
+    Set the virtual pkg module if the os is AIX
+    '''
+    if __grains__['os_family'] == 'AIX':
+        return __virtualname__
+    return (False,
+            'The aixpkg execution module failed to load.')
```
Comment (cachedout) [inline salt/modules/aixpkg.py]: I don't think this is a good error message. `failed` suggests that it should have loaded and did not. Perhaps something like `Did not load AIX module on non-AIX OS`?

File: salt/modules/aixpkg.py [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+    .. code-block:: bash
+
+        salt '*' pkg._is_installed bash
+    '''
+    cmd = ['/usr/bin/lslpp', '-Lc', name]
+    return __salt__['cmd.retcode'](cmd) == 0
+
+
+def install(name=None, refresh=False, pkgs=None, version=None, test=False, **kwargs):
```
Comment (cachedout) [inline salt/modules/aixpkg.py]: `version` and `test` doesn't appear to be documented below.

Comment (dmurphy18) [reply salt/modules/aixpkg.py]: @cachedout Will update to include refresh, version and test.

**Comment by rallytime** [comment]
@dmurphy18 I have changed the base branch of this PR from develop to fluorine for inclusion in the Fluorine feature release.

**Review by terminalmage — APPROVED** [review]

**Pull request merged**